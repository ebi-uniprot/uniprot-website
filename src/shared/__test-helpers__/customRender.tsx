/**
 * This tries to follow what is detailed here:
 * https://testing-library.com/docs/react-testing-library/setup#custom-render
 */
import { render, RenderOptions } from '@testing-library/react';
import { Component, Dispatch, ReactElement, ReactNode } from 'react';
import { HelmetProvider } from 'react-helmet-async';
import {
  createMemoryHistory,
  LocationState,
  MemoryHistory,
  Route,
  Router,
} from 'react-router';
import { JsonValue, SetRequired } from 'type-fest';

import { MessagesState } from '../../messages/state/messagesInitialState';
import { MessagesAction } from '../../messages/state/messagesReducers';
import databaseInfo from '../../uniprotkb/utils/__tests__/__mocks__/databaseInfo';
import { getDatabaseInfoMaps } from '../../uniprotkb/utils/database';
import {
  MessagesDispatchContext,
  MessagesStateContext,
} from '../contexts/Messages';
import {
  DatabaseInfoMapsContext,
  UniProtDataVersionContext,
} from '../contexts/UniProtData';
import { JobAction } from '../workers/jobs/state/actionHandler';
import { JobsState } from '../workers/jobs/state/jobsInitialState';

type ExtraRenderOptions = {
  // For react-router
  history?: MemoryHistory<LocationState>;
  path?: string;
  // For custom user preferences (used by useLocalStorage)
  initialLocalStorage?: Record<string, JsonValue>;
  // For app context
  messagesState?: MessagesState;
  messagesDispatch?: jest.Mock<Dispatch<MessagesAction>>;
  JobsState?: JobsState;
  toolsDispatch?: jest.Mock<Dispatch<JobAction>>;
};

type WrapperProps = { children: ReactNode } & RenderOptions &
  SetRequired<
    ExtraRenderOptions,
    | 'history'
    | 'initialLocalStorage'
    | 'messagesState'
    | 'messagesDispatch'
    | 'JobsState'
    | 'toolsDispatch'
  >;

const dbInfoMaps = getDatabaseInfoMaps(databaseInfo);
const uniProtDataVersion = {
  releaseNumber: '2023_01',
  releaseDate: new Date('01-March-2023'),
};

class Wrapper extends Component<WrapperProps> {
  constructor(props: WrapperProps) {
    super(props);
    for (const [key, value] of Object.entries(props.initialLocalStorage)) {
      window.localStorage.setItem(key, JSON.stringify(value));
    }
  }

  // eslint-disable-next-line class-methods-use-this
  componentWillUnmount() {
    window.localStorage.clear();
  }

  render() {
    const { children, path, history, messagesDispatch, messagesState } =
      this.props;
    return (
      <HelmetProvider>
        <Router history={history}>
          <MessagesDispatchContext.Provider value={messagesDispatch}>
            <MessagesStateContext.Provider value={messagesState}>
              <DatabaseInfoMapsContext.Provider value={dbInfoMaps}>
                <UniProtDataVersionContext.Provider value={uniProtDataVersion}>
                  {path ? (
                    <Route path={path} render={() => children} />
                  ) : (
                    children
                  )}
                </UniProtDataVersionContext.Provider>
              </DatabaseInfoMapsContext.Provider>
            </MessagesStateContext.Provider>
          </MessagesDispatchContext.Provider>
        </Router>
      </HelmetProvider>
    );
  }
}

const customRender = (
  ui: ReactElement,
  {
    route = '',
    path,
    history = createMemoryHistory({ initialEntries: [route] }),
    initialLocalStorage = {},
    messagesState = {},
    messagesDispatch = jest.fn(),
    JobsState = {},
    toolsDispatch = jest.fn(),
    ...options
  }: RenderOptions &
    ExtraRenderOptions & {
      route?: string;
    } = {}
) => ({
  ...render(ui, {
    wrapper: (props) => (
      <Wrapper
        path={path}
        history={history}
        initialLocalStorage={initialLocalStorage}
        messagesState={messagesState}
        messagesDispatch={messagesDispatch}
        JobsState={JobsState}
        toolsDispatch={toolsDispatch}
        {...props}
      />
    ),
    ...options,
  }),
  history,
  messagesDispatch,
  toolsDispatch,
});

export default customRender;
