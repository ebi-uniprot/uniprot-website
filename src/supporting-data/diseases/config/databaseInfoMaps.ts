export type DiseasesDatabaseInfoPoint = {
  name: string;
  displayName: string;
  uriLink: string;
};

export type DiseasesDatabaseToDatabaseInfo = {
  [database: string]: DiseasesDatabaseInfoPoint;
};

const allDatabases: DiseasesDatabaseInfoPoint[] = [
  {
    name: 'MedGen',
    displayName: 'MedGen',
    uriLink: 'https://www.ncbi.nlm.nih.gov/medgen/%id',
  },
  {
    name: 'MeSH',
    displayName: 'MeSH',
    uriLink: 'https://meshb.nlm.nih.gov/record/ui?ui=%id',
  },
  {
    name: 'MIM',
    displayName: 'MIM',
    uriLink: 'https://www.omim.org/entry/%id',
  },
];

const databaseToDatabaseInfo: DiseasesDatabaseToDatabaseInfo =
  Object.fromEntries(allDatabases.map((db) => [db.name, db]));

export default databaseToDatabaseInfo;
