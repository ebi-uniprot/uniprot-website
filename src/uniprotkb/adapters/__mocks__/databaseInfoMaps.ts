import { DatabaseInfoMaps } from '../../utils/database';
import databaseCategoryToNames from './databaseCategoryToNames';
import databaseNameToCategory from './databaseNameToCategory';
import databaseToDatabaseInfo from './databaseToDatabaseInfo';
import entrySectionToDatabaseCategoryOrder from './entrySectionToDatabaseCategoryOrder';
import entrySectionToDatabaseNames from './entrySectionToDatabaseNames';
import implicitDatabaseXRefs from './implicitDatabaseXrefs';

/**
 * This is generated, based on data from '/configure/uniprotkb/allDatabases'
 */

const databaseInfoMaps: DatabaseInfoMaps = {
  databaseCategoryToNames,
  databaseNameToCategory,
  databaseToDatabaseInfo,
  implicitDatabaseXRefs,
  entrySectionToDatabaseNames,
  entrySectionToDatabaseCategoryOrder,
};

export default databaseInfoMaps;
