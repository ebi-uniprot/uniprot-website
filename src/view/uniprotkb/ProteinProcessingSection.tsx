import React, { FC } from 'react';
import { Card } from 'franklin-sites';
import hasContent from '../../model/utils/utils';
import EntrySection from '../../model/types/EntrySection';
import FeaturesView from './components/FeaturesView';
import KeywordView from './components/KeywordView';
import XRefView from './components/XRefView';
import FreeTextView from './components/FreeTextView';
import { CommentType, FreeText } from '../../model/types/CommentTypes';
import { UIModel } from '../../model/uniprotkb/SectionConverter';

const ProteinProcessingSection: FC<{
  data: UIModel;
  primaryAccession: string;
  sequence: string;
}> = ({ data, sequence, primaryAccession }): JSX.Element | null => {
  if (!hasContent(data)) {
    return null;
  }
  const { featuresData, keywordData, xrefData, commentsData } = data;
  return (
    <div id={EntrySection.ProteinProcessing}>
      <Card title={EntrySection.ProteinProcessing}>
        <FeaturesView features={featuresData} sequence={sequence} />
        <FreeTextView
          comments={commentsData.get(CommentType.PTM) as FreeText[]}
          includeTitle
        />
        <KeywordView keywords={keywordData} />
        <XRefView xrefs={xrefData} primaryAccession={primaryAccession} />
      </Card>
    </div>
  );
};

export default ProteinProcessingSection;
