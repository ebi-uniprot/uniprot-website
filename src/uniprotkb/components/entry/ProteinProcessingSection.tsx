import { Card, Loader } from 'franklin-sites';

import EntrySection, {
  getEntrySectionNameAndId,
} from '../../types/entrySection';
import FeaturesView from '../protein-data-views/UniProtKBFeaturesView';
import KeywordView from '../protein-data-views/KeywordView';
import XRefView from '../protein-data-views/XRefView';
import FreeTextView from '../protein-data-views/FreeTextView';

import useDataApi from '../../../shared/hooks/useDataApi';

import { hasContent } from '../../../shared/utils/utils';
import { proteinsApi } from '../../../shared/config/apiUrls';
import { convertPtmexchangeFeatures } from '../../adapters/ptmexchangeFeaturesConverter';

import { FreeTextComment } from '../../types/commentTypes';
import { ProteomicsPtm } from '../../types/proteomicsPtm';
import { UIModel } from '../../adapters/sectionConverter';

type Props = {
  data: UIModel;
  primaryAccession: string;
  sequence: string;
};

const ProteinProcessingSection = ({
  data,
  sequence,
  primaryAccession,
}: Props) => {
  const { loading: proteomicsPtmLoading, data: proteomicsPtmData } =
    useDataApi<ProteomicsPtm>(proteinsApi.proteomicsPtm(primaryAccession));

  if (proteomicsPtmLoading) {
    return <Loader />;
  }

  const convertedPtmexchangeFeatures = proteomicsPtmData
    ? convertPtmexchangeFeatures(proteomicsPtmData.features)
    : [];

  if (!hasContent(data) || !convertedPtmexchangeFeatures.length) {
    return null;
  }
  const { featuresData, keywordData, xrefData, commentsData } = data;
  return (
    <Card
      header={
        <h2 data-article-id="ptm_processing_section">
          {getEntrySectionNameAndId(EntrySection.ProteinProcessing).name}
        </h2>
      }
      id={EntrySection.ProteinProcessing}
      data-entry-section
    >
      <FeaturesView
        primaryAccession={primaryAccession}
        features={[...featuresData, ...convertedPtmexchangeFeatures]}
        sequence={sequence}
        showSourceColumn={!!convertedPtmexchangeFeatures.length}
      />
      <FreeTextView
        comments={commentsData.get('PTM') as FreeTextComment[] | undefined}
        articleId="post-translational_modification"
        title="Post-translational modification"
      />
      <KeywordView keywords={keywordData} />
      <XRefView xrefs={xrefData} primaryAccession={primaryAccession} />
    </Card>
  );
};

export default ProteinProcessingSection;
