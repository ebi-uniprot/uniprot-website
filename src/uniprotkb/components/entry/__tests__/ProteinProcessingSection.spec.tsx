import { screen } from '@testing-library/react';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import customRender from '../../../../shared/__test-helpers__/customRender';
import uniProtKbConverter from '../../../adapters/uniProtkbConverter';
import ProteinProcessingSection from '../ProteinProcessingSection';

import mockHumanData from '../../../__mocks__/uniProtKBEntryModelData';
import mockPtmExchangeData from '../../../__mocks__/ptmExchangeData';
import mockNonHumanData from '../../../__mocks__/nonHumanEntryModelData';
import databaseInfoMaps from '../../../utils/__tests__/__mocks__/databaseInfoMaps';
import EntrySection from '../../../types/entrySection';

const axiosMock = new MockAdapter(axios);
axiosMock
  // Humans should (eventually) have PTMeXchange data so just using
  .onGet(
    `https://www.ebi.ac.uk/proteins/api/proteomics-ptm/${mockHumanData.primaryAccession}`
  )
  .reply(200, mockPtmExchangeData)
  // Assumed that flavonifractor plautii won't have PTMeXchange data
  .onGet(
    `https://www.ebi.ac.uk/proteins/api/proteomics-ptm/${mockNonHumanData.primaryAccession}`
  )
  .reply(404);

describe('ProteinProcessingSection', () => {
  it('should render when PTMeXchange is available', async () => {
    const transformedData = uniProtKbConverter(mockHumanData, databaseInfoMaps);
    const { asFragment } = customRender(
      <ProteinProcessingSection
        data={transformedData[EntrySection.ProteinProcessing]}
        sequence={transformedData[EntrySection.Sequence].sequence.value}
        primaryAccession={transformedData.primaryAccession}
        key={EntrySection.ProteinProcessing}
      />,
      { route: `/uniprotkb/P05067/entry` }
    );
    await screen.findByText('PTM/Processing');
    await screen.findByText('Download');
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render when no PTMeXchange is available', async () => {
    const transformedData = uniProtKbConverter(
      mockNonHumanData,
      databaseInfoMaps
    );
    const { asFragment } = customRender(
      <ProteinProcessingSection
        data={transformedData[EntrySection.ProteinProcessing]}
        sequence={transformedData[EntrySection.Sequence].sequence.value}
        primaryAccession={transformedData.primaryAccession}
        key={EntrySection.ProteinProcessing}
      />,
      { route: `/uniprotkb/P05067/entry` }
    );
    await screen.findByText('PTM/Processing');
    await screen.findByText('Download');
    expect(asFragment()).toMatchSnapshot();
  });
});
