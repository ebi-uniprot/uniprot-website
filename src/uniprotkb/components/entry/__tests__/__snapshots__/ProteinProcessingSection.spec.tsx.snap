// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProteinProcessingSection should render when PTMeXchange is available 1`] = `
<DocumentFragment>
  <section
    class="card"
    data-entry-section="true"
    id="ptm_processing"
  >
    <div
      class="card__container"
    >
      <div
        class="card__header card__header--with-separator"
      >
        <h2
          data-article-id="ptm_processing_section"
        >
          PTM/Processing
        </h2>
      </div>
      <div
        class="card__content"
      >
        <h3>
          Features
        </h3>
        <p>
          Showing features for 
          <span
            data-article-id="chain"
          >
            chain
          </span>
          , 
          <span
            data-article-id="mod_res_large_scale"
          >
            modified residue (large scale data)
          </span>
          .
        </p>
        <protvista-manager
          attributes="highlight displaystart displayend selectedid"
        >
          <protvista-zoom-tool
            length="10"
          >
            <span
              class="nightingale-button-content"
              slot="zoom-in"
            >
              <test-file-stub
                height="19"
              />
            </span>
            <span
              class="nightingale-button-content"
              slot="zoom-out"
            >
              <test-file-stub
                height="19"
              />
            </span>
            <span
              class="nightingale-button-content"
              slot="zoom-in-seq"
            >
              <test-file-stub
                height="19"
              />
            </span>
          </protvista-zoom-tool>
          <button
            class="button tertiary"
            type="button"
          >
            <test-file-stub />
            Download
          </button>
          <a
            class="full-view"
            href="/uniprotkb/undefined/feature-viewer"
            title="View in the Feature Viewer"
          >
            <test-file-stub
              height="19"
            />
          </a>
          <protvista-navigation
            length="10"
          />
          <protvista-track
            layout="non-overlapping"
            length="10"
          />
          <protvista-sequence
            height="20"
            length="10"
            sequence="SAPSQDFMRF"
          />
          <div
            class="datatable-wrapper"
          >
            <protvista-datatable
              filter-scroll="true"
            >
              <table>
                <thead>
                  <tr>
                    <th
                      data-filter="type"
                    >
                      Type
                    </th>
                    <th>
                      ID
                    </th>
                    <th>
                      Position(s)
                    </th>
                    <th
                      data-filter="source"
                    >
                      Source
                    </th>
                    <th>
                      Description
                    </th>
                    <th />
                  </tr>
                </thead>
                <tbody>
                  <tr
                    data-end="8"
                    data-id="id value CHAIN"
                    data-start="2"
                  >
                    <td
                      data-filter="type"
                      data-filter-value="Chain"
                    >
                      Chain
                    </td>
                    <td
                      id="id value CHAIN"
                    >
                      id value CHAIN
                    </td>
                    <td>
                      2-8
                    </td>
                    <td
                      data-filter="source"
                      data-filter-value="UniProt"
                    >
                      UniProt
                    </td>
                    <td
                      id="description2-8"
                    >
                      Cu
                      <sup>
                        2+
                      </sup>
                       A2 (
                      <a
                        href="/uniprotkb?query=ft_binding:"ChEBICHEBI:3314""
                      >
                        UniProtKB
                      </a>
                       | 
                      <a
                        class="external-link"
                        href="https://www.ebi.ac.uk/chebi/searchId.do?chebiId=ChEBICHEBI:3314"
                        rel="noopener"
                        target="_blank"
                      >
                        ChEBI
                        <test-file-stub
                          data-testid="external-link-icon"
                          width="12.5"
                        />
                      </a>
                      ); Some note of Ca
                      <sup>
                        2+
                      </sup>
                       A1 (
                      <a
                        href="/uniprotkb?query=ft_binding:"ChEBICHEBI:3214""
                      >
                        UniProtKB
                      </a>
                       | 
                      <a
                        class="external-link"
                        href="https://www.ebi.ac.uk/chebi/searchId.do?chebiId=ChEBICHEBI:3214"
                        rel="noopener"
                        target="_blank"
                      >
                        ChEBI
                        <test-file-stub
                          data-testid="external-link-icon"
                          width="12.5"
                        />
                      </a>
                      ); Some note; In isoform 
                      <a
                        href="/#Isoform_sequence"
                      >
                        sequence
                      </a>
                       1; description value 123
                      <button
                        aria-controls="1"
                        aria-expanded="false"
                        class="svg-colour-reviewed evidence-tag"
                        data-testid="evidence-tag-trigger"
                        type="button"
                      >
                        <test-file-stub
                          height="12"
                          width="12"
                        />
                        <span
                          class="evidence-tag__label"
                        >
                          1 publication
                        </span>
                      </button>
                      <div
                        class="evidence-tag-content "
                        data-testid="evidence-tag-content"
                        id="1"
                      />
                    </td>
                    <td>
                      <div
                        class="button-group"
                      >
                        <a
                          class="button tertiary"
                          href="/blast?ids=P21802[2-8]"
                          title="BLAST the sequence corresponding to this feature"
                          translate="no"
                        >
                          BLAST
                        </a>
                        <button
                          class="button tertiary no-small"
                          title="Add 1 entry to the basket"
                          type="button"
                        >
                          <test-file-stub />
                          Add
                        </button>
                      </div>
                    </td>
                  </tr>
                  <tr
                    data-end="8"
                    data-group-for="id value CHAIN"
                    data-start="2"
                  >
                    <td>
                      <strong>
                        Sequence: 
                      </strong>
                      original value â†’ alternative value
                    </td>
                  </tr>
                  <tr
                    data-end="215"
                    data-id="0"
                    data-start="215"
                  >
                    <td
                      data-filter="type"
                      data-filter-value="Modified residue (large scale data)"
                    >
                      Modified residue (large scale data)
                    </td>
                    <td />
                    <td>
                      215
                    </td>
                    <td
                      data-filter="source"
                      data-filter-value="PTMeXchange"
                    >
                      PTMeXchange
                    </td>
                    <td
                      id="description215"
                    >
                      Phosphoserine
                      <span
                        data-article-id="mod_res_large_scale#what-is-the-goldsilverbronze-criterion"
                      >
                        <button
                          class="chip chip--compact secondary Silver"
                          type="button"
                        >
                          Silver
                        </button>
                      </span>
                      <button
                        aria-controls="2"
                        aria-expanded="false"
                        class="svg-colour-unreviewed evidence-tag"
                        data-testid="evidence-tag-trigger"
                        type="button"
                      >
                        <test-file-stub
                          height="12"
                          width="12"
                        />
                        <span
                          class="evidence-tag__label"
                        >
                          Combined sources
                        </span>
                      </button>
                      <div
                        class="evidence-tag-content "
                        data-testid="evidence-tag-content"
                        id="2"
                      />
                    </td>
                    <td />
                  </tr>
                </tbody>
              </table>
            </protvista-datatable>
            <button
              class="button primary toggle-button"
              type="button"
            >
              Expand table
            </button>
          </div>
        </protvista-manager>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`ProteinProcessingSection should render when no PTMeXchange is available 1`] = `
<DocumentFragment>
  <section
    class="card"
    data-entry-section="true"
    id="ptm_processing"
  >
    <div
      class="card__container"
    >
      <div
        class="card__header card__header--with-separator"
      >
        <h2
          data-article-id="ptm_processing_section"
        >
          PTM/Processing
        </h2>
      </div>
      <div
        class="card__content"
      >
        <h3>
          Features
        </h3>
        <p>
          Showing features for 
          <span
            data-article-id="signal"
          >
            signal
          </span>
          , 
          <span
            data-article-id="chain"
          >
            chain
          </span>
          .
        </p>
        <protvista-manager
          attributes="highlight displaystart displayend selectedid"
        >
          <protvista-zoom-tool
            length="772"
          >
            <span
              class="nightingale-button-content"
              slot="zoom-in"
            >
              <test-file-stub
                height="19"
              />
            </span>
            <span
              class="nightingale-button-content"
              slot="zoom-out"
            >
              <test-file-stub
                height="19"
              />
            </span>
            <span
              class="nightingale-button-content"
              slot="zoom-in-seq"
            >
              <test-file-stub
                height="19"
              />
            </span>
          </protvista-zoom-tool>
          <button
            class="button tertiary"
            type="button"
          >
            <test-file-stub />
            Download
          </button>
          <a
            class="full-view"
            href="/uniprotkb/undefined/feature-viewer"
            title="View in the Feature Viewer"
          >
            <test-file-stub
              height="19"
            />
          </a>
          <protvista-navigation
            length="772"
          />
          <protvista-track
            layout="non-overlapping"
            length="772"
          />
          <protvista-sequence
            height="20"
            length="772"
            sequence="MRNRRKAVSLLTGLLVTAQLFPTAALAADSSESALNKAPGYQDFPAYYSDSAHADDQVTHPDVVVLEEPWNGYRYWAVYTPNVMRISIYENPSIVASSDGVHWVEPEGLSNPIEPQPPSTRYHNCDADMVYNAEYDAMMAYWNWADDQGGGVGAEVRLRISYDGVHWGVPVTYDEMTRVWSKPTSDAERQVADGEDDFITAIASPDRYDMLSPTIVYDDFRDVFILWANNTGDVGYQNGQANFVEMRYSDDGITWGEPVRVNGFLGLDENGQQLAPWHQDVQYVPDLKEFVCISQCFAGRNPDGSVLHLTTSKDGVNWEQVGTKPLLSPGPDGSWDDFQIYRSSFYYEPGSSAGDGTMRVWYSALQKDTNNKMVADSSGNLTIQAKSEDDRIWRIGYAENSFVEMMRVLLDDPGYTTPALVSGNSLMLSAETTSLPTGDVMKLETSFAPVDTSDQVVKYTSSDPDVATVDEFGTITGVSVGSARIMAETREGLSDDLEIAVVENPYTLIPQSNMTATATSVYGGTTEGPASNVLDGNVRTIWHTNYAPKDELPQSITVSFDQPYTVGRFVYTPRQNGTNGIISEYELYAIHQDGSKDLVASGSDWALDAKDKTVSFAPVEAVGLELKAIAGAGGFGTAAELNVYAYGPIEPAPVYVPVDDRDASLVFTGAWNSDSNGSFYEGTARYTNEIGASVEFTFVGTAIRWYGQNDVNFGAAEVYVDGVLAGEVNVYGPAAAQQLLFEADGLAYGKHTIRIVCVSPVVDFDYFSYVGE"
          />
          <div
            class="datatable-wrapper"
          >
            <protvista-datatable
              filter-scroll="true"
            >
              <table>
                <thead>
                  <tr>
                    <th
                      data-filter="type"
                    >
                      Type
                    </th>
                    <th>
                      ID
                    </th>
                    <th>
                      Position(s)
                    </th>
                    <th>
                      Description
                    </th>
                    <th />
                  </tr>
                </thead>
                <tbody>
                  <tr
                    data-end="27"
                    data-id="0"
                    data-start="1"
                  >
                    <td
                      data-filter="type"
                      data-filter-value="Signal"
                    >
                      Signal
                    </td>
                    <td />
                    <td>
                      1-27
                    </td>
                    <td
                      id="description1-27"
                    >
                      <button
                        aria-controls="1"
                        aria-expanded="false"
                        class="svg-colour-reviewed evidence-tag"
                        data-testid="evidence-tag-trigger"
                        type="button"
                      >
                        <test-file-stub
                          height="12"
                          width="12"
                        />
                        <span
                          class="evidence-tag__label"
                        >
                          Sequence analysis
                        </span>
                      </button>
                      <div
                        class="evidence-tag-content "
                        data-testid="evidence-tag-content"
                        id="1"
                      />
                    </td>
                    <td>
                      <div
                        class="button-group"
                      >
                        <a
                          class="button tertiary"
                          href="/blast?ids=P0DTR4[1-27]"
                          title="BLAST the sequence corresponding to this feature"
                          translate="no"
                        >
                          BLAST
                        </a>
                        <button
                          class="button tertiary no-small"
                          title="Add 1 entry to the basket"
                          type="button"
                        >
                          <test-file-stub />
                          Add
                        </button>
                      </div>
                    </td>
                  </tr>
                  <tr
                    data-end="27"
                    data-group-for="0"
                    data-start="1"
                  >
                    <td>
                      <strong>
                        Sequence: 
                      </strong>
                      MRNRRKAVSLLTGLLVTAQLFPTAALA
                    </td>
                  </tr>
                  <tr
                    data-end="772"
                    data-id="PRO_0000448571"
                    data-start="28"
                  >
                    <td
                      data-filter="type"
                      data-filter-value="Chain"
                    >
                      Chain
                    </td>
                    <td
                      id="PRO_0000448571"
                    >
                      PRO_0000448571
                    </td>
                    <td>
                      28-772
                    </td>
                    <td
                      id="A_type_blood_N-acetyl-alpha-D-galactosamine_deacetylase"
                    >
                      A type blood N-acetyl-alpha-D-galactosamine deacetylase
                    </td>
                    <td>
                      <div
                        class="button-group"
                      >
                        <a
                          class="button tertiary"
                          href="/blast?ids=P0DTR4[28-772]"
                          title="BLAST the sequence corresponding to this feature"
                          translate="no"
                        >
                          BLAST
                        </a>
                        <button
                          class="button tertiary no-small"
                          title="Add 1 entry to the basket"
                          type="button"
                        >
                          <test-file-stub />
                          Add
                        </button>
                      </div>
                    </td>
                  </tr>
                  <tr
                    data-end="772"
                    data-group-for="PRO_0000448571"
                    data-start="28"
                  >
                    <td>
                      <strong>
                        Sequence: 
                      </strong>
                      ADSSESALNKAPGYQDFPAYYSDSAHADDQVTHPDVVVLEEPWNGYRYWAVYTPNVMRISIYENPSIVASSDGVHWVEPEGLSNPIEPQPPSTRYHNCDADMVYNAEYDAMMAYWNWADDQGGGVGAEVRLRISYDGVHWGVPVTYDEMTRVWSKPTSDAERQVADGEDDFITAIASPDRYDMLSPTIVYDDFRDVFILWANNTGDVGYQNGQANFVEMRYSDDGITWGEPVRVNGFLGLDENGQQLAPWHQDVQYVPDLKEFVCISQCFAGRNPDGSVLHLTTSKDGVNWEQVGTKPLLSPGPDGSWDDFQIYRSSFYYEPGSSAGDGTMRVWYSALQKDTNNKMVADSSGNLTIQAKSEDDRIWRIGYAENSFVEMMRVLLDDPGYTTPALVSGNSLMLSAETTSLPTGDVMKLETSFAPVDTSDQVVKYTSSDPDVATVDEFGTITGVSVGSARIMAETREGLSDDLEIAVVENPYTLIPQSNMTATATSVYGGTTEGPASNVLDGNVRTIWHTNYAPKDELPQSITVSFDQPYTVGRFVYTPRQNGTNGIISEYELYAIHQDGSKDLVASGSDWALDAKDKTVSFAPVEAVGLELKAIAGAGGFGTAAELNVYAYGPIEPAPVYVPVDDRDASLVFTGAWNSDSNGSFYEGTARYTNEIGASVEFTFVGTAIRWYGQNDVNFGAAEVYVDGVLAGEVNVYGPAAAQQLLFEADGLAYGKHTIRIVCVSPVVDFDYFSYVGE
                    </td>
                  </tr>
                </tbody>
              </table>
            </protvista-datatable>
            <button
              class="button primary toggle-button"
              type="button"
            >
              Expand table
            </button>
          </div>
        </protvista-manager>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
